src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -417,16 +534,113 @@ g.test('vertexAccess')
     pass.setPipeline(pipeline);
 
     // Run the draw variant
-    draw.insertInto(pass, p.indexed, p.indirect);
+    drawCall.insertInto(pass, isIndexed, isIndirect);
 
     pass.endPass();
-    t.device.queue.submit([encoder.finish()]);
+    this.device.queue.submit([encoder.finish()]);
 
-    // Validate we see green instead of red, meaning no fragment ended up on-screen
-    t.expectSinglePixelIn2DTexture(
+    // Validate we see green on the left pixel, showing that no failure case is detected
+    this.expectSinglePixelIn2DTexture(
       colorAttachment,
       'rgba8unorm',
       { x: 0, y: 0 },
       { exp: new Uint8Array([0x00, 0xff, 0x00, 0xff]), layout: { mipLevel: 0 } }
     );
+    // Validate wee see red on the right pixel, showing that at least one case succeed, prevent the test
+    // from unexpectly doing nothing and pass the validation
+    this.expectSinglePixelIn2DTexture(
+      colorAttachment,
+      'rgba8unorm',
+      { x: 1, y: 0 },
+      { exp: new Uint8Array([0xff, 0x00, 0x00, 0xff]), layout: { mipLevel: 0 } }
+    );
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('vertex_buffer_access')
+  .params(u =>
+    u
+      .combineWithParams([
+        { indexed: false, indirect: true },
+        { indexed: true, indirect: false },
+        { indexed: true, indirect: true },
+      ])
+      .expand('drawCallTestParameter', function* (p) {
+        if (p.indexed) {
+          yield* ['baseVertex', 'vertexCountInIndexBuffer'] as const;
+          if (p.indirect) {
+            yield* ['indexCount', 'instanceCount', 'firstIndex', 'firstInstance'] as const;
+          }
+        } else {
+          if (p.indirect) {
+            yield* ['vertexCount', 'instanceCount', 'firstVertex', 'firstInstance'] as const;
+          }
+        }
+      })
+      .combine('type', Object.keys(typeInfoMap))
+      .combine('additionalBuffers', [0, 4])
+      .combine('partialLastNumber', [false, true])
+      .combine('offsetVertexBuffer', [false, true])
+      .combine('errorScale', [0, 1, 4, 10 ** 2, 10 ** 4, 10 ** 6])
+  )
+  .fn(async t => {
+    const p = t.params;
+    const typeInfo = typeInfoMap[p.type];
+
+    // Number of vertices to draw
+    const numVertices = 4;
+    // Each buffer will be bound to this many attributes (2 would mean 2 attributes per buffer)
+    const attributesPerBuffer = 2;
+    // Random values to fill our buffer with to avoid collisions with other tests
+    const arbitraryValues = Array(5)
```

Comment:
I have strong concerns about using actual randomness in tests: if it passes in 99% of the cases but not in the other then we'll see flakiness that's really hard to debug. @kainino0x WDYT of providing some helper functions for that, that maybe hash the query string to produce a random seed? (we'd still need to avoid Math.random because of the global state I think)

Reply:
I replace the random values with some arbitrary values I choose.